### ISSUE TITLE
Compiling with C++ Program

### ISSUE THREAD

**Issue by avickars** [NVIDIA/VideoProcessingFramework#504]
**Describe the bug**

I am trying to use VPF in c++ and having trouble linking everything together properly.  To be succinct, my sample program is super basic I am just trying to get it to compile:

```
#include <iostream>
#include <NvDecoder.h>

int main() {
    std::cout << "Hello, World!" << std::endl;
    return 0;
}
```

Since I don't need the python bindings, I am only compiling the source code in the `/home/VideoProcessingFramework/src/TC` directory via:

```
cd VideoProcessingFramework/src/TC
mkdir build
cmake ..
make
```

And to compile my program Im using the following makefile:

```
SHELL := /bin/bash
CC=g++ -std=c++17 -Wall -g
NV=nvcc

PKGS:= libavcodec libavfilter libavformat libavutil libswresample

CFLAGS:= -I/home/VideoProcessingFramework/src/TC/inc/
CFLAGS+= -I/home/VideoProcessingFramework/src/TC/TC_CORE/inc/
CFLAGS+= -I/home/VideoProcessingFramework/src/TC/build/TC_CORE/
CFLAGS+= -I/home/VideoProcessingFramework/src/TC/third_party/
CFLAGS+= -I/usr/include/linux/
CFLAGS+= -I/home/VideoProcessingFramework/src/TC/build/
CFLAGS+= $(shell pkg-config --cflags $(PKGS))

LIBS:= $(shell pkg-config --libs $(PKGS))
LIBS+= -lTC_CORE -lTC

LINK:= -L/home/VideoProcessingFramework/src/TC/build/TC_CORE/
LINK+= -L/home/VideoProcessingFramework/src/TC/build/

all:
	$(CC) test.cpp -o test $(CFLAGS) $(LIBS) $(LINK)
```

This gives a whole bunch of errors so I am clear missing something.  Any help on what actually needs to be included/linked together would be greatly appreciated!

As an FYI, I am using the `nvcr.io/nvidia/tritonserver:22.12-py3` as a base docker image.  My docker file is in a comment below.  Also, when I execute `pip install .` everything works no problem, it really just appears to be a linking issue where Im not getting everything that needs to be linked.

**To Reproduce**

**Expected behavior**
It should compile.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Linux
 - Nvidia driver version: 520.61.05
 - CUDA Version 11.8
 - Python Version 3.8
 - Video Codec SDK Version (only if using a custom header from the Video Codec SDK via the CMake variable `TC_VIDEO_CODEC_INTERFACE_DIR`)

**Additional context**

**Comment by avickars** [comment]
Dockerfile

```
FROM nvcr.io/nvidia/tritonserver:22.12-py3

RUN apt-get update

# ****************************************** pkg-config ******************************************
RUN apt-get install -y pkg-config \
    ffmpeg
RUN pip install cmake --upgrade


# ****************************************** VPF ******************************************
RUN apt install -y \
              libavfilter-dev \
              libavformat-dev \
              libavcodec-dev \
              libswresample-dev \
              libavutil-dev\
              wget \
              build-essential \
              git

WORKDIR /home
RUN git clone https://github.com/NVIDIA/VideoProcessingFramework.git
WORKDIR ./VideoProcessingFramework/src/TC
```

**Comment by RomanArzumanyan** [comment]
Hi @avickars 

You can simply link your application against `TC_CORE` and `TC` libraries. You can follow `PyNvCodec` sources as sample illustrating how to use C++ interface.

**Comment by avickars** [comment]
Hi @RomanArzumanyan ,

Thank you for your response!

I totally understand that PyNvCodec is linked against `TC` etc.  My confusion is on actually linking with the `TC` and `TC_Core` libraries.  I can see that after compiling the `TC` and `TC_Core` libraries, the static libraries `/home/VideoProcessingFramework/src/TC/build/libTC.a` and `/home/VideoProcessingFramework/src/TC/build/TC_CORE/libTC_CORE.a` are generated and I have adjusted my makefile to include them.  However, when I compile my test program that links to these libraries I am still getting a multitude of errors and I have been unable to determine what I am missing.

In short, is there any sample c++ program that just shows the compilation when linked to the `TC` and `TC_CORE` libraries.  To be clear Im not looking for any program that shows how the libraries work in c++ as that can be easily seen by just looking at the python bindings.  Im simply just stuck getting my program to compile as from what understand there isn't any reference.

Thanks so much! Any help is appreciated!

**Comment by RomanArzumanyan** [comment]
@avickars 

Ah, I see. There were changes to TC/CMakeLists.txt:
https://github.com/NVIDIA/VideoProcessingFramework/blob/3347e555ed795ba7de98b4e6b9bf7fe441784663/src/TC/CMakeLists.txt#L40-L53

It's no longer shared library target. Probably that was done to simplify the `pip install` process.

Unless it's properly fixed I assume you can do a quick dirty fix (if you don't need anything Python) by adding `SHARED` keyword to `TC` and `TC_CORE` CMakeLists.txt files.

**Comment by avickars** [comment]
Hey @RomanArzumanyan ,

I tried that and unfortunately it didn't work.  To be clear I added `SHARED` to: https://github.com/NVIDIA/VideoProcessingFramework/blob/3347e555ed795ba7de98b4e6b9bf7fe441784663/src/TC/TC_CORE/CMakeLists.txt#L27 and https://github.com/NVIDIA/VideoProcessingFramework/blob/3347e555ed795ba7de98b4e6b9bf7fe441784663/src/TC/CMakeLists.txt#L40

and Im still not able to compile and am getting a multitude of errors.  For debugging purposes I uploaded the output from the compile attempt here: https://drive.google.com/file/d/1SZu-X4oTVDTx3R0EsdGKgpGM4eiaELb7/view?usp=sharing

**Comment by avickars** [comment]
update: Got it to work (stupid mistake in my makefile).  Thanks for your help @RomanArzumanyan !
